secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/internationalpayments/InternationalPaymentsApiController.java

```diff
@@ -0,0 +1,278 @@
+import com.forgerock.sapi.gateway.ob.uk.rs.validation.obie.v4.payment.OBWriteInternational3Validator.OBWriteInternational3ValidationContext;
+import com.forgerock.sapi.gateway.rcs.consent.store.client.payment.international.InternationalPaymentConsentStoreClient;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.ConsumePaymentConsentRequest;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.international.v3_1_10.InternationalPaymentConsent;
+import com.forgerock.sapi.gateway.rs.resource.store.repo.entity.payment.FRInternationalPaymentSubmission;
+import com.forgerock.sapi.gateway.rs.resource.store.repo.mongo.payments.InternationalPaymentSubmissionRepository;
+import com.forgerock.sapi.gateway.uk.common.shared.api.meta.obie.OBVersion;
+
+import jakarta.servlet.http.HttpServletRequest;
+import uk.org.openbanking.datamodel.v4.common.Meta;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteInternational3;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteInternationalResponse5;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteInternationalResponse5Data;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetails1;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetails1Status;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetails1StatusDetail;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetails1StatusDetailStatus;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetailsResponse1;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetailsResponse1Data;
+
+@Controller("InternationalPaymentsApiV4.0.0")
+public class InternationalPaymentsApiController implements InternationalPaymentsApi {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final InternationalPaymentSubmissionRepository paymentSubmissionRepository;
+    private final PaymentSubmissionValidator paymentSubmissionValidator;
+    private final InternationalPaymentConsentStoreClient consentStoreClient;
+    private final OBValidationService<OBWriteInternational3ValidationContext> paymentValidator;
+    private final RefundAccountService refundAccountService;
+    private final IdempotentPaymentService<FRInternationalPaymentSubmission, FRWriteInternational> idempotentPaymentService;
+
+    public InternationalPaymentsApiController(
+            InternationalPaymentSubmissionRepository paymentSubmissionRepository,
+            PaymentSubmissionValidator paymentSubmissionValidator,
+            @Qualifier("v4.0.0RestInternationalPaymentConsentStoreClient") InternationalPaymentConsentStoreClient consentStoreClient,
+            OBValidationService<OBWriteInternational3ValidationContext> paymentValidator,
+            RefundAccountService refundAccountService) {
+        this.paymentSubmissionRepository = paymentSubmissionRepository;
+        this.paymentSubmissionValidator = paymentSubmissionValidator;
```

Comment:
Null checks needed here?

Reply:
Already checked as this is a replica of the prev version (validators, services, etc.).